network-protection/network-protection-api/src/main/java/com/duckduckgo/networkprotection/api/NetworkProtectionState.kt

```diff
@@ -48,4 +50,16 @@ interface NetworkProtectionState {
      * @return Returns the server location if available, otherwise null.
      */
     fun serverLocation(): String?
+
+    /**
+     * This method returns the [ConnectionState] of Network Protection which considers both the feature state and the VPNService state.
+     * @return [ConnectionState] of Network Protection.
+     */
+    fun getConnectionStateFlow(): Flow<ConnectionState>
```

Comment:
Note for myself. I hate that we're forced to do this because of how our settings are structured 😢

Reply:
Yeah, I tried to think of more clever ways to do this but this is the best I came up with.